### PR TITLE
Troubleshoot Helm error when using Azure Issuer

### PR DESCRIPTION
**Pull request by MrSimonEmms**
Fixes https://github.com/cert-manager/cert-manager/issues/4867

There's a particularly confusing message when using an Azure Issuer in Helm with the AKS Kubelet identity. This is just something that requires documenting how to get around.

### CODE REVIEW

**Comment by MrSimonEmms** [comment]
I've tried to maintain the style. Happy to make amends with guidance if necessary

**Review by irbekrm — CHANGES_REQUESTED** [review]
Thank you for this PR, adding this information to the docs was long needed!

I have reworded it to include explanation why `Issuer`s in particular need the flag set + removed Helm reference- let me know if that makes sense, also the comment needs to be moved up to pod identities section I think.

File: content/en/docs/configuration/acme/dns01/azuredns.md [hunk 1]
```diff
@@ -213,6 +213,15 @@ spec:
...
 
+When using an `Issuer` with the Helm charts, you may see this error:
+
+```bash
+error instantiating azuredns challenge solver: ClientID is not set but neither --cluster-issuer-ambient-credentials nor --issuer-ambient-credentials are set.
+These are necessary to enable Azure Managed Identities
+```
+
+To solve this, please add `--issuer-ambient-credentials=true` to the `extraArgs` section.
+
```
[the remark this answers is not in the document]
Comment (MrSimonEmms) [reply content/en/docs/configuration/acme/dns01/azuredns.md:224]: HI @irbekrm, thanks for that. I'll make the changes and resubmit it. I like the new wording as it explains the problem better than I could.

You're right about the other authentication mechanisms not needing this as they don't use ambient credentials.

**Comment by MrSimonEmms** [comment]
Updated PR to include @irbekrm's suggestion and improved wording. I've changed a couple of stylistic things and a couple of typos from their suggested changed, but mostly imported as-is

**Comment by irbekrm** [comment]
Thanks for making the changes @MrSimonEmms !

Just one more small nit- can we move this addition to [the pod identities section](https://github.com/cert-manager/website/blob/3ecebce6e09c7cf031396f4b7fcbd7b5ec93dc70/content/en/docs/configuration/acme/dns01/azuredns.md#managed-identity-using-aad-pod-identities) (so around line 136)?
At the moment it has been added to [kubelet identity section](https://cert-manager.io/docs/configuration/acme/dns01/azuredns/#managed-identity-using-aks-kubelet-identity) which should not require ambient credentials flag

**Comment by MrSimonEmms** [comment]
@irbekrm that's not a nit at all - I'd gone code-blind and put it in the wrong section. Good spot

**Comment by irbekrm** [comment]
Awesome, this now looks great to me!

Thanks again for taking the initiative to document this bit!

/ok-to-test
/lgtm
/approve

**Pull request merged**